src/grid.jl

```diff
@@ -0,0 +1,90 @@
+# Grid types
+export Node, Cell, Grid
+
+# Cell typealias
+export Line, QuadraticLine,
+       Triangle, QuadraticTriangle, Quadrilateral, QuadraticQuadrilateral,
+       Tetrahedron, QuadraticTetrahedron, Hexahedron, QuadraticHexahedron
+
+# Grid utilities
+export cells, nodes, cell_coordinates, n_cells, n_nodes
+
+#########################
+# Main types for meshes #
+#########################
+"""
+A `Node` is a point in space.
+"""
+immutable Node{dim, T}
+    x::Vec{dim, T}
+end
+
+"""
+A `Cell` is a sub-domain defined by a collection of `Node`s as vertices.
+"""
+immutable Cell{dim, N}
+    id::Int
```

Comment:
You mean **cell** 1?

I'm just thinking what datatype `cell` should be when looping over `for cell in cells(mesh)`? Perhaps a `CellIndex` where
```
immutable CellIndex
    idx::Int
end
```

so we can dispatch on the `cell` compared to for example a `BoundaryIndex` or something...